fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1603,31 +1609,38 @@ public static IndexState writeOnlyIfTooManyRecordsForRebuild(long recordCount, b
         tr.clear(getSubspace().range(Tuple.from(INDEX_UNIQUENESS_VIOLATIONS_KEY)));
         List<CompletableFuture<Void>> work = new LinkedList<>();
         addRebuildRecordCountsJob(work);
-        return rebuildIndexes(getRecordMetaData().getIndexesSince(-1), Collections.emptyMap(), work, RebuildIndexReason.REBUILD_ALL, null);
+        return rebuildIndexes(getRecordMetaData().getIndexesSince(-1), Collections.emptyMap() /* new states */, work, RebuildIndexReason.REBUILD_ALL, null);
     }
 
+    /**
+     * Prepare an index for rebuilding by clearing any existing data and marking the index as write-only.
+     * @param indexName the name of the index to build
+     * @param reason the rebuild reason
+     * @return a future that completes when the index has been cleared and marked write-only for building
+     */
     @Nonnull
-    public CompletableFuture<Void> clearAndMarkIndexWriteOnly(@Nonnull String indexName) {
-        return clearAndMarkIndexWriteOnly(metaDataProvider.getRecordMetaData().getIndex(indexName));
+    public CompletableFuture<Void> clearAndMarkIndexWriteOnly(@Nonnull String indexName, @Nonnull RebuildIndexReason reason) {
```

Comment:
This is changing a public API of a `STABLE` interface, so you cannot break the API in this way.  You'll need to do something like:
 

    /**
      * ...
      * @deprecated see {@link #clearAndMarkIndexWriteOnly(String, IndexRebuildReason)
      */
    @API(API.Status.DEPRECATED)
    @Deprecated
    @Nonnull
    public CompletableFuture<Void> clearAndMarkIndexWriteOnly(@Nonnull String indexName) {
        return clearAndMarkIndexWriteOnly(indexName, RebuildIndexReason.UNKNOWN);
    }

    public CompletableFuture<Void> clearAndMarkIndexWriteOnly(@Nonnull String indexName, @Nonnull RebuildIndexReason reason)

Reply:
done.